salt/modules/napalm_mod.py

```diff
@@ -1015,8 +1015,19 @@ def pyeapi_nxos_api_args(**prev_kwargs):
     kwargs["username"] = napalm_opts["USERNAME"]
     kwargs["password"] = napalm_opts["PASSWORD"]
     kwargs["timeout"] = napalm_opts["TIMEOUT"]
-    kwargs["transport"] = optional_args.get("transport")
-    kwargs["port"] = optional_args.get("port")
+    # kwargs["transport"] = optional_args.get("transport", "https")
+    # kwargs["port"] = optional_args.get("port", 443)
+
+    if "transport" in optional_args and optional_args["transport"]:
```

Comment:
**suggestion**:
`kwargs["transport'} = options_args.get("transport", "https)`

Reply:
using get("key") is generally in case the key doesn't exist, but the code already just checked that it is, hence can use the indexed method.  Part of the previous problem was the code was finding the 'transport' key (which was None) and just returning that, and using get("key") will do the same with giving you None

```
david@david-laptop2:~/isos/redhat$ python3
Python 3.8.5 (default, Oct 23 2020, 09:35:03) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> dict = {}
>>> dict["transport"] = None
>>> mydog = dict.get("transport", "https")
>>> if mydog:
...     print(f"mydog has '{mydog}'")
... else:
...     print("mydog is empty or None")
... 
mydog is empty or None
>>> 
```

Reply:
Will clean out those left over comments above too